### PR TITLE
Add `upgrade-temp-backup` directory in `wp-content`

### LINKED ISSUE #287: [Bug] add`upgrade-temp-backup` creation to salt to allow WordPress theme and plugin updates
## Description

When attempting a plugin or theme update via `wpcli` the following error occurs: 

`````
wpcli plugin update PLUGINNAME
plugin update PLUGINNAME
Enabling Maintenance mode...
Downloading update from https://downloads.wordpress.org/plugin/PLUGINNAME.x.x.x.zip...
Unpacking the update...
Installing the latest version...
Warning: Could not create the upgrade-temp-backup directory.
Plugin update failed.
Disabling Maintenance mode...
+------------+-------------+-------------+--------+
| name       | old_version | new_version | status |
+------------+-------------+-------------+--------+
| PLUGINNAME | x.x.x       | x.x.x       | Error  |
+------------+-------------+-------------+--------+
Error: No plugins updated (1 failed).
`````

[Due to a change from WordPress v6.3 onwards](https://make.wordpress.org/core/2023/07/11/new-in-6-3-rollback-for-failed-manual-plugin-and-theme-updates/), when performing upgrades to plugins or themes the following directories must be present and properly permissioned: 

* `wp-content/upgrade-temp-backup/plugins/`
* `wp-content/upgrade-temp-backup/themes/`

Therefore, we should ensure that the `upgrade-temp-backup` folder is present. I am unsure if we also need to precreate and permission the `plugins` and `themes` directories as well. I believe the creation of the `upgrade-temp-backup` folder alone with the right user and group permissions should suffice; but I leave that up to other minds.

## Reproduction

1. run `wpcli plugin update PLUGINNAME`
4. See error.

## Expectation
The update should run successfully without error.

## Resolution

- [ ] I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by Shafiya-Heena**
## Fixes
- Fixes #287  by @Shafiya-Heena 

## Description
- update init.sls to create `upgrade-temp-backup` dir in `wp-content`
- update init.sls to create `plugins` and `themes` in `upgrade-temp-backup` dir

## Tests
- tested on `index__stage__us-east-2`

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

For the purposes of this DCO, "license" is equivalent to "license or public domain dedication," and "open source license" is equivalent to "open content license or public domain dedication."
<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: states/wordpress/init.sls [hunk 1]
```diff
@@ -133,6 +133,29 @@ include:
       - composer: {{ sls }} composer update
 
 
+{{ sls }} dir wp-content/upgrade-temp-back:
+    file.directory:
+      - name: {{ DOCROOT }}/wp-content/upgrade-temp-back
```
Comment (possumbilities) [inline states/wordpress/init.sls:138]: the folder name here is `upgrade-temp-back` when it should be `upgrade-temp-backup`

File: states/wordpress/init.sls [hunk 2]
```diff
@@ -133,6 +133,29 @@ include:
...
+      - group: webdev
+      - require:
+        - file: {{ sls }} dir wp-content
+      - require_in:
+        - composer: {{ sls }} composer update
+
+{%- for dir in ["plugins", "themes"] %}
+{{ sls }} dir wp-content/upgrade-temp-back/{{ dir }}:
+    file.directory:
+      - name: {{ DOCROOT }}/wp-content/upgrade-temp-back/{{ dir }}
```
Comment (possumbilities) [inline states/wordpress/init.sls:149]: the folder here is `upgrade-temp-back` when it should be `upgrade-temp-backup`

File: states/wordpress/init.sls [hunk 3]
```diff
@@ -133,6 +133,29 @@ include:
...
+      - name: {{ DOCROOT }}/wp-content/upgrade-temp-back
+      - mode: '2775'
+      - group: webdev
+      - require:
+        - file: {{ sls }} dir wp-content
+      - require_in:
+        - composer: {{ sls }} composer update
+
+{%- for dir in ["plugins", "themes"] %}
+{{ sls }} dir wp-content/upgrade-temp-back/{{ dir }}:
```
Comment (possumbilities) [inline states/wordpress/init.sls:147]: the folder here is `upgrade-temp-back` when it should be `upgrade-temp-backup`

File: states/wordpress/init.sls [hunk 4]
```diff
@@ -133,6 +133,29 @@ include:
...
+    file.directory:
+      - name: {{ DOCROOT }}/wp-content/upgrade-temp-back/{{ dir }}
+      - mode: '2775'
+      - group: webdev
+      - require:
+        - file: {{ sls }} dir wp-content/upgrade-temp-back
```
Comment (possumbilities) [inline states/wordpress/init.sls:153]: the folder here is `upgrade-temp-back` when it should be `upgrade-temp-backup`

File: states/wordpress/init.sls (hunk 1 above)
Comment (Shafiya-Heena) [reply states/wordpress/init.sls]: Good catch! I've updated it. Thank you! 👍

**Review by possumbilities — APPROVED** [review]
From my own view, this looks sound. Approved.

**Pull request merged**